src/linter.cc

```diff
@@ -20,94 +20,305 @@
+
+    if (location >= sql.size() || sql[location] != ';') {
+      return absl::Status(absl::StatusCode::kFailedPrecondition,
+                          "Each statemnt should end with a consequtive"
+                          "semicolon ';'");
     }
+  }
+  return absl::OkStatus();
+}
 
-    return return_status;
+bool AllUpperCase(const absl::string_view &sql,
+                  const ParseLocationRange &range) {
+  for (int i = range.start().GetByteOffset(); i < range.end().GetByteOffset();
+       ++i) {
+    if ('a' <= sql[i] && sql[i] <= 'z') return false;
+  }
+  return true;
 }
 
-absl::Status checkStatement(absl::string_view sql) {
-    std::unique_ptr<zetasql::ParserOutput> output;
-    return zetasql::ParseStatement(sql,
-        zetasql::ParserOptions(), &output);
+absl::Status CheckUppercaseKeywords(absl::string_view sql) {
+  ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
+  std::vector<ParseToken> parse_tokens;
+  absl::Status tokenizer_status =
+      GetParseTokens(ParseTokenOptions(), &location, &parse_tokens);
+
+  // Keyword definition in tokenizer is very wide,
+  // it include some special characters like ';', '*', etc.
+  // Keyword Uppercase check will simply ignore characters
+  // outside of english lowercase letters.
+  for (auto &token : parse_tokens) {
+    if (token.kind() == ParseToken::KEYWORD) {
+      if (!AllUpperCase(sql, token.GetLocationRange())) {
+        return absl::Status(
+            absl::StatusCode::kFailedPrecondition,
+            absl::StrCat("All keywords should be Uppercase, In character ",
+                         std::to_string(
+                             token.GetLocationRange().start().GetByteOffset()),
+                         " string should be: ", token.GetSQL()));
+      }
+    }
+  }
+
+  return absl::OkStatus();
 }
 
+absl::Status CheckCommentType(absl::string_view sql, char delimeter) {
+  bool dash_comment = false;
+  bool slash_comment = false;
+  bool hash_comment = false;
+  bool inside_string = false;
+
+  for (int i = 0; i < static_cast<int>(sql.size()); ++i) {
+    if (sql[i] == '\'' || sql[i] == '"') inside_string = !inside_string;
+
+    if (inside_string) continue;
 
-absl::Status checkLineLength(absl::string_view sql, int lineLimit,
-    const char delimeter) {
-    int lineSize = 0;
-    int lineNumber = 1;
-    for (int i=0; i<static_cast<int>(sql.size()); i++) {
-        if ( sql[i] == delimeter ) {
-            lineSize = 0;
-            lineNumber++;
-        } else {
-            lineSize++;
-        }
-        if ( lineSize > lineLimit ) {
-            return absl::Status(
-                absl::StatusCode::kFailedPrecondition,
-                absl::StrCat("Lines should be <= ", std::to_string(lineLimit),
-                " characters long [", std::to_string(lineNumber), ",1]") );
-        }
+    if (i > 0 && sql[i - 1] == '-' && sql[i] == '-') {
+      dash_comment = true;
+      // ignore the line.
```

Comment:
nit: start a comment with capital letter. https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar
(you already mastered adding a dot in the end :))